tests/generators/test_rest.py

```diff
@@ -15,11 +15,13 @@
 
 @pytest.fixture
 def set_rest_config():
-    _config.plugins.generators["rest.RestGenerator"] = {
+    _config.plugins.generators["rest"] = {}
+    _config.plugins.generators["rest"]["RestGenerator"] = {
         "name": DEFAULT_NAME,
         "uri": DEFAULT_URI,
+        "api_key": "testing",
+        "generations": DEFAULT_GENERATIONS_QTY,
     }
-    _config.run.generations = DEFAULT_GENERATIONS_QTY
```

Comment:
This pattern works if the goal is to test generator-specific run quantity configuration. The intent of the original test is to test compliance with the overall run quantity. Happy to test both compliance with the overall generations qty and invocation-specific quantities, but I'd prefer to retain the test of compliance with the global value also.

Reply:
This test file is for the `rest` module, the ability to override `generations` via `_config.run` should be considered out of scope here. We can implement an integration test in `test_cli` or `test_config` that provides for validation of compliance with the global override.

Reply:
Thanks for the clarity. So - does the recommended change in this test set the `generations` value on the generator instance, or the global value in `_config.run`?

Reply:
It sets it on the `_config.plugins.generators` key that is passed as the default by the constructor, since the `_config.run` value ends up propagated to the same place in `cli.py`.

Future work may completely decouple `_config.run` values from plugin instantiation and expect the caller to provide a fully propagated configuration specific to how the loaded instance should look for the object being created.

Reply:
OK, this will be a change to keep an eye on!